js/coinsbit.js

```diff
@@ -0,0 +1,520 @@
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostAccountBalances (params);
+        const balances = this.safeValue (response, 'result');
+        const currencies = Object.keys (balances);
+        const parsedBalances = {};
+        const free = {};
+        const used = {};
+        const total = {};
+        for (let currencyIndex = 0; currencyIndex < currencies.length; currencyIndex++) {
+            const currency = currencies[currencyIndex];
+            const balance = balances[currency];
+            const parsedBalance = {
+                'free': this.safeFloat (balance, 'available'),
+                'used': this.safeFloat (balance, 'freeze'),
+                'total': this.safeFloat (balance, 'available') + this.safeFloat (balance, 'freeze'),
+            };
+            parsedBalances[currency] = parsedBalance;
+            free[currency] = parsedBalance['free'];
+            used[currency] = parsedBalance['used'];
+            total[currency] = parsedBalance['total'];
+        }
+        parsedBalances['free'] = free;
+        parsedBalances['used'] = used;
+        parsedBalances['total'] = total;
+        parsedBalances['info'] = balances;
+        return parsedBalances;
+    }
+
+    async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrder requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            symbol['limit'] = limit;
+        }
+        if (since !== undefined) {
+            symbol['since'] = limit;
+        }
+        const response = await this.privatePostOrders (this.extend (request, params));
+        const orders = this.safeValue (this.safeValue (response, 'result'), 'result');
+        return this.parseOrders (orders, market, since, limit, params);
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrder requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const response = await this.privatePostAccountOrderHistory (params);
+        const result = this.safeValue (response, 'result');
+        if (this.isArray (result)) {
+            // User has no closed orders yet.
+            return [];
+        }
+        const market = this.market (symbol);
+        const marketId = market['id'];
+        if (marketId in result) {
+            const orders = this.safeValue (result, market['id']);
+            return this.parseOrders (orders, market, since, limit, params);
+        } else {
+            return [];
+        }
```

Comment:
The above 7 lines can be rewritten as follows:

```JavaScript
const marketId = market['id'];
        if (marketId in result) {
            const orders = this.safeValue (result, market['id']);
            return this.parseOrders (orders, market, since, limit, params);
        } else {
            return [];
        }
```

↓

```JavaScript
const orders = this.safeValue (result, market['id'], []);
return this.parseOrders (orders, market, since, limit); // no need to pass the request 'params' here
```